samza-kv/src/main/scala/org/apache/samza/storage/kv/BaseKeyValueStorageEngineFactory.scala

```diff
@@ -162,4 +184,28 @@ trait BaseKeyValueStorageEngineFactory[K, V] extends StorageEngineFactory[K, V]
       keyValueStorageEngineMetrics, batchSize, () => clock.nanoTime())
   }
 
+  def createCachedStore[K, V](enableCache: Boolean, storeName: String, registry: MetricsRegistry,
```

Comment:
Instead of passing a boolean as a method parameter, it's better to just do the if-else at the call site and call/not call the method.

Reply:
Done.